src/dataServices/dynamoDbBundleService.ts

```diff
@@ -190,7 +193,11 @@ export class DynamoDbBundleService implements Bundle {
         const idItemsFailedToRead: string[] = [];
         for (let i = 0; i < itemResponses.length; i += 1) {
             const itemResponse = itemResponses[i];
-            if (itemResponse instanceof ResourceNotFoundError) {
+            // allow for update as create scenario
+            if (
+                itemResponse instanceof ResourceNotFoundError &&
+                !(itemsToLock[i].operation === 'update' && this.updateCreateSupported)
+            ) {
                 idItemsFailedToRead.push(`${itemsToLock[i].resourceType}/${itemsToLock[i].id}`);
```

Comment:
So allowing `updateAsCreate` what is the versionId created? will it take what ever is in the json or will the server generate it?

Reply:
I don't think there's a way for the resource in a bundle to provide a vid. If it had one it would just be ignored.
If the vid can't be found in the `idToVersionId` map (which is populated with the data coming back from lockItems) it's assumed to be 1: https://github.com/awslabs/fhir-works-on-aws-persistence-ddb/blob/56c16a44692b892a042baa92b53ccc4e661ae146/src/dataServices/dynamoDbBundleServiceHelper.ts#L66